modules/apps/portal-security/portal-security-permission-impl/src/main/java/com/liferay/portal/security/permission/internal/InlineSQLHelperImpl.java

```diff
@@ -134,6 +134,10 @@ public boolean isEnabled(long companyId, long groupId) {
 			throw new IllegalStateException("Permission checker is null");
 		}
 
+		if (companyId == 0) {
```

Comment:
Hi @yuhai and @dantewang quick question on this implementation, why do you prefer this fix rather than d7cfcfdf2b3b86d6e71b3526b3e826136257a9ad? 
In any case, 0 or greater > 0, the companyId should be matched with the value on this call: `permissionChecker.getCompanyId()`

Reply:
Hi @Vy-CES, yes, most cases permissionChecker.getCompanyId() > 0. We consider PermissionChecker may be custom implementation. In custom implementation, it may be 0 and it override isOmniadmin() method. For example, we have DummyPermissionChecker in the test util module. So we choose to use relatively small changes to fix the issue.

Reply:
@yuhai thanks for the reply,

is my understanding correct that in this case, the companyId is always > 0, and when that happens (companyId>0) the `else if `with the condition` (permissionChecker.isCompanyAdmin(companyId))` is an equivalent to `permissionChecker.isOmniadmin()`?

Reply:
@Vy-CES, in the case, just let the logic goes to `permissionChecker.isCompanyAdmin(companyId)` to fix the issue. They aren't equivalent between `isCompanyAdmin` and `isOmniadmin`

Reply:
@yuhai please consider the following test cases, regardless of companyId and/or groupId, the last **else** condition will never be reached. 
```
companyId = 0, groupId > 0 ---> first if.
companyId = 0, groupId = 0 ---> else if.
companyId > 0, groupId = 0 ---> else if.
```
CC: @dantewang @tinatian